src/ert/analysis/_es_update.py

```diff
@@ -191,25 +191,16 @@ def _save_temporary_storage_to_disk(
 ) -> None:
     for key, matrix in temporary_storage.items():
         config_node = ensemble_config.getNode(key)
-
-        if isinstance(config_node, GenKwConfig):
-            parameter_keys = list(ensemble_config.get_keyword_model_config(key))
-            target_fs.save_gen_kw(
-                key,
-                parameter_keys,
-                iens_active_index,
-                matrix,
-            )
-        elif isinstance(config_node, SurfaceConfig):
-            for i, realization in enumerate(iens_active_index):
+        for i, realization in enumerate(iens_active_index):
+            if isinstance(config_node, GenKwConfig):
+                parameter_keys = list(ensemble_config.get_keyword_model_config(key))
```

Comment:
Does this make sense? Isn't config_node already `ensemble_config.get_keyword_model_config(key)` ?

```
    def get_keyword_model_config(self, key: str) -> GenKwConfig:
        return self[key]
```

Reply:
Good point, this should be equivalent to:

```python

if isinstance(config_node, GenKwConfig):
                parameter_keys = list(config_node)
```